### PR TITLE
Updates action to node20

### LINKED ISSUE #33: Update action to use `node@20`
GitHub recently added support for `node@20` when creating JavaScript based GitHub actions. We should update the action metadata to ensure that we support the latest version of node.

[docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#example-using-nodejs-v20](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#example-using-nodejs-v20)

### PR DESCRIPTION
**Pull request by milanholemans**
* Fixed a few typos
* Updated dependencies
* Bumped runtime node version to node20
* Bumped GH workflow actions to node20

### CODE REVIEW

**Review by garrytrinder — APPROVED** [review]

**Comment by garrytrinder** [comment]
Great work @milanholemans 👍

**Comment by milanholemans** [comment]
@garrytrinder updated some extra dependencies of the workflow. Could you have a look again?

File: README.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node-version: [18.x]
+        node-version: [20.x]
```
Comment (waldekmastykarz) [inline README.md]: Since SPFx doesn't support Node@20 yet, we should stick to 18 here.

File: README.md [hunk 2]
```diff
@@ -105,7 +105,7 @@ jobs:
...
```
Comment (waldekmastykarz) [inline README.md]: Since SPFx doesn't support Node@20 yet, we should stick to 18 here.

File: README.md [hunk 3]
```diff
@@ -147,7 +147,7 @@ jobs:
...
```
Comment (waldekmastykarz) [inline README.md]: Since SPFx doesn't support Node@20 yet, we should stick to 18 here.

**Review by waldekmastykarz — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (milanholemans) [reply README.md]: I was a little bit too fast there indeed 😊

**Review by waldekmastykarz — APPROVED** [review]

**Pull request merged**